docs/config.md
# Settings
The following are a list of user settings that are common to all event generators. Note that the input key is case-insensitive.

## General Settings

- **Generators**: A list of generators {cite}`Alwall:2014hca,Bahr:2008pv,Kilian:2007gr,Sherpa:2024mfk,Bierlich:2022pfr,Jadach:2022mbe,CarloniCalame:2000pz,Campbell:2022qmc,Bellm:2015jjp` whose runcards should be generated. One generator must be specified from the currently supported set of:

```yaml
Generators:
  - Babayaga
  - Herwig
  - Madgraph
  - Pythia
  - Sherpa
  - Whizard

```
- **Model**: Name of the model (default: SM)

- **SqrtS**: $\sqrt{s}$ in GeV.

- **ISRMode**: Enable Initial State Radiation (ISR):
```yaml
ISRMode: 1
```
default: 0 (turned off)

- **RandomSeed**: Seed for the event generation

- **OutputFormat**: Format in which the monte-carlo events will be written out to. Currently supported options are edm4hep, hepmc and lhe (default: edm4hep)

- **OutDir**: The directory to which the run-cards will be saved. Current default is $PWD/Run-Cards.

- **Events**: Number of Monte-Carlo events to be generated.

- **EWParamDevThreshold**: threshold for issueing a warning if in the EW parameters at LO a relative deviation is larger

- **NLO**: integer to turn on/off NLO with default: 0 (off)

- **Processes**: A list of processes which runcards should be generated. Each process should have its own unique name. Under these headings you can
				 specify the initial and final states, a list of pdg codes, to be generated and at what order e.g [EW,QCD].
```yaml
Processes:
  Muon:
     Initial: [11, -11]
     Final: [13, -13]
  MuonNeutrino:
     Initial: [11, -11]
     Final: [14, -14]
  Tau:
     Initial: [11, -11]
     Final: [15, -15]
  TauNeutrino:
     Initial: [11, -11]
     Final: [16, -16]
```

## Particle Data

Particle properties can be modified with respect to the default:

- **ParticleData**: The user can set various particle properties such as mass and width. Note it is expected of the user to set a consistent input scheme.
					The particles are identified using the corresponding PDG number

```yaml
ParticleData:
  25:
    mass: 125
    width: 0
  11:
    mass: 0.0005111
    width: 0
  23:
    mass: 91.1876
    width: 2.4952
  24:
    mass: 80.379
    width: 2.085
```

## Phase Space

By default, no restrictions are applied on the available phase space. If necessary:
- **Selectors**: Some basic one and two particle phasespace cuts can be set. Each will need to be set with a minimum and maximum value as well as the flavour(s)
				it should be applied to.
  - **One Particle Selectors**:
  	- **PT**: Cut on the transverse momenta
  	- **ETA**: Cut on the pseduorapidity
  	- **Y**: Cut on the Rapidity
  	- **ET**: Cut on the Transverse energy

  - **Two Particle Selectors**:
  	- **Mass**: Cut on the invariant mass of two particles
  	- **Angle**: Cut on the angular separation in radians
  	- **DEta**: Cut on the pseudorapidity separation
  	- **DY**: Cut on the rapidity separation
  	- **DPhi**: Cut on the azimuthal separation in radians
  	- **DR**: Cut on the R separation


## Generator Specific Settings
Here we summarise the settings available to a subset of generators.


- **Beam Polarization**: Polarized beams are available for Madgraph, Sherpa and Whizard. It can be set as follows
```yaml
PositronPolarisation: 0.3
ElectronPolarisation: 0.8
```
- **Beamstrahlung**: is turned on by specifying the type of accelerator. Allowed values are: ILC, FCC, C3, CEPC, HALFHF.
```yaml
Beamstrahlung: ILC
```
Note that **Beamstrahlung** is conditional on **ISRmode** being on. The **Beamstrahlung** and **SqrtS** variables are used to configure the settings of the generators. For **SqrtS** a vicinity search (within 10GeV) is performed. In case the requested setting does not exist, a replacement setting is used and printed as a warning.

## Analysis
Postgeneration analyses can be performed either using Rivet {cite}`Bierlich:2024vqo, Bierlich:2019rhm` and/or key4hep {cite}`Ganis:2021vgv, Key4hep:2022xly`
```
Analysis:
  Tools: [key4hep, rivet]
  RivetAnalysis: [MC_XS, MC_ZINC,...]
  RivetPath: /path/to/analysis
```


_sources/config.md.txt
# General Settings
The following are a list of user settings that are common to all event generators. Note that the input key is case-insensitive.

- **Generators**: A list of generators {cite}`Alwall:2014hca,Kilian:2007gr,Sherpa:2024mfk,Bierlich:2022pfr,Jadach:2022mbe,CarloniCalame:2000pz,Campbell:2022qmc,Bellm:2015jjp` whose runcards should be generated. One generator must be specified
```yaml
Generators:
  - Sherpa
  - Whizard
  - Madgraph

```
- **Model**: Name of the model (default: SM)

- **SqrtS**: $\sqrt{s}$ in GeV.

- **ISRMode**: Enable ISR via electron structure function:
```yaml
ISRMode: 1
```
default: 0 (turned off)

- **RandomSeed**: Seed for the event generation

- **OutputFormat**: Format in which the monte-carlo events will be written out to. Currently supported options are edm4hep, hepmc and lhe (default: edm4hep)

- **OutDir**: The directory to which the run-cards will be saved. Current default is $PWD/Run-Cards.

- **Events**: Number of Monte-Carlo events to be generated.

- **EWParamDevThreshold**: threshold for issueing a warning if in the EW parameters at LO a relative deviation is larger

- **NLO**: integer to turn on/off NLO with default: 0 (off)

- **Processes**: A list of processes which runcards should be generated. Each process should have its own unique name. Under these headings you can
				 specify the final states to be generated and at what order e.g [EW,QCD].
```yaml
Processes:
  Muon:
     Initial: [11, -11]
     Final: [13, -13]
  MuonNeutrino:
     Initial: [11, -11]
     Final: [14, -14]
  Tau:
     Initial: [11, -11]
     Final: [15, -15]
  TauNeutrino:
     Initial: [11, -11]
     Final: [16, -16]
```

- **ParticleData**: Here the user can set various particle properties such as mass and width. Note it is expected of the user to set a consistent input scheme.
					The particles are identified using the corresponding PDG number

```yaml
ParticleData:
  25:
    mass: 125
    width: 0
  11:
    mass: 0.0005111
    width: 0
  23:
    mass: 91.1876
    width: 2.4952
  24:
    mass: 80.379
    width: 2.085
```

- **Selectors**: Some basic one and two particle phasespace cuts can be set. Each will need to be set with a minimum and maximum value as well as the flavour(s)
				it should be applied to.
  - **One Particle Selectors**:
  	- **PT**: Cut on the transverse momenta
  	- **ETA**: Cut on the pseduorapidity
  	- **Y**: Cut on the Rapidity
  	- **ET**: Cut on the Transverse energy

  - **Two Particle Selectors**:
  	- **Mass**: Cut on the invariant mass of two particles
  	- **Angle**: Cut on the angular separation in radians
  	- **DEta**: Cut on the pseudorapidity separation
  	- **DY**: Cut on the rapidity separation
  	- **DPhi**: Cut on the azimuthal separation in radians
  	- **DR**: Cut on the R separation


## Generator Specific Settings
Here we summarise the settings available to a subset of generators.


- **Beam Polarization**: Polarized beams are available for Madgraph and Whizard. It can be set as follows
```yaml
PositronPolarisation: 0.3
ElectronPolarisation: 0.8
```
- **Beamstrahlung**: is turned on by specifying the type of accelerator. Allowed values are: ILC, FCC, C3, CEPC, HALFHF.
```yaml
Beamstrahlung: ILC
```
Note that **Beamstrahlung** is conditional on **ISRmode** being on. The **Beamstrahlung** and **SqrtS** variables are used to configure the settings of the generators. For **SqrtS** a vicinity search (within 10GeV) is performed. In case the requested setting does not exist, a replacement setting is used and printed as a warning.

## Analysis
Postgeneration analyses can be performed either using Rivet {cite}`Bierlich:2024vqo, Bierlich:2019rhm` and/or key4hep {cite}`Ganis:2021vgv, Key4hep:2022xly`
```
Analysis:
  Tools: [key4hep, rivet]
  RivetAnalysis: [MC_XS, MC_ZINC,...]
  RivetPath: /path/to/analysis
```


python/Parameters.py
# Based on Feynrules and UFO formats


class Parameter:
    require_args = ["name", "value", "isParticleProperty"]

    def __init__(self, name, value, isParticleProperty):
        args = (name, value, isParticleProperty)
        for i, prop in enumerate(self.require_args):
            setattr(self, prop, args[i])
        # keep a global list of parameters
        self.updateList(name)

    @staticmethod
    def set_info(name, value):
        for _, v in globals().items():
            if isinstance(v, Parameter) and v.name == name:
                v.value = value

    @staticmethod
    def get_info(name):
        for _, v in globals().items():
            if isinstance(v, Parameter) and v.name == name:
                return v
        raise ValueError(f"Could not find Parameter with name {name}")

    @staticmethod
    def updateList(name):
        for o,v in globals().items():
            if o=="ParametersList" and isinstance(v, list):
                v.append(name)

# list of required parameters
ParametersList = []

#default parameter and mass values

alphaEMMZM1 = Parameter(
    name="alphaEMMZM1",
    value=127.9,
    isParticleProperty=False
)

alphaEMMZ = Parameter(
    name="alphaEMMZ",
    value=1/alphaEMMZM1.value,
    isParticleProperty=False
)

alphaEMLOM1 = Parameter(
    name="alphaEMLOM1",
    value=1.32184e+02,
    isParticleProperty=False
)

alphaEMLO = Parameter(
    name="alphaEMLO",
    value=1/alphaEMLOM1.value,
    isParticleProperty=False
)

alphaEMM1 = Parameter(
    name="alphaEMM1",
    value=137.035999139,
    isParticleProperty=False
)

alphaEM = Parameter(
    name="alphaEM",
    value=1/alphaEMM1.value,
    isParticleProperty=False
)

GFermi = Parameter(
    name="GFermi",
    value=0.0000116637,
    isParticleProperty=False
)

sin2thetaLO = Parameter(
    name="sin2thetaLO",
    value=0.223013,
    isParticleProperty=False
)

sin2theta = Parameter(
    name="sin2theta",
    value=0.23155,
    isParticleProperty=False
)

sin2thetaEff = Parameter(
    name="sin2thetaEff",
    value=0.23155,
    isParticleProperty=False
)

alphaSMZ = Parameter(
    name="alphaSMZ",
    value=0.1184,
    isParticleProperty=False
)

VEV = Parameter(
    name="VEV",
    value=246,
    isParticleProperty=False
)

MZ = Parameter(
    name="MZ",
    value=91.1876,
    isParticleProperty=True
)

WZ = Parameter(
    name="WZ",
    value=2.4952,
    isParticleProperty=True
)

MW = Parameter(
    name="MW",
    value=80.379,
    isParticleProperty=True
)

WW = Parameter(
    name="WW",
    value=2.085,
    isParticleProperty=True
)

MB = Parameter(
    name="MB",
    value=4.7,
    isParticleProperty=True
)

ymb = Parameter(
    name="ymb",
    value=MB.value/VEV.value,
    isParticleProperty=True
)

MT = Parameter(
    name="MT",
    value=172,
    isParticleProperty=True
)

WT = Parameter(
    name="WT",
    value=1.50833649,
    isParticleProperty=True
)

ymt = Parameter(
    name="ymt",
    value=MT.value/VEV.value,
    isParticleProperty=True
)

MH = Parameter(
    name="MH",
    value=125,
    isParticleProperty=True
)

WH = Parameter(
    name="WH",
    value=0.00407,
    isParticleProperty=True
)


MU_R = Parameter(
    name="MU_R",
    value=91.188,
    isParticleProperty=False
)

